### ISSUE TITLE
Migrate RHOAR QE coverage for MicroProfile OpenTracing

### ISSUE THREAD

**Issue by rsvoboda** [jboss-eap-qe/eap-microprofile-test-suite#10]
Migrate relevant test coverage from 

 - https://github.com/rhoar-qe/thorntail-test-suite/tree/master/microprofile

**Comment by zhantemirov** [comment]
I've been waiting a while for the output https://github.com/jboss-eap-qe/eap-microprofile-test-suite/pull/20#pullrequestreview-313214564 using test-containers tooling proposal. 

For now it looks undefined so I had to migrate Docker helper classes too in order to get the migrated tests to work and tune them up. I had some difficulties with swarm @DefaultDeployment annotation eliminating. Deployments are OK now but Jaeger seems to be not grabbing logs from Tracer in deployed service - I will have to have a look on it next week once I am back in the office.

Due to the above mentioned reasons the working [branch](https://github.com/zhantemirov/eap-microprofile-test-suite/tree/migrate-open-tracing) is still in progress.

**Comment by mnovak1** [comment]
@zhantemirov I can probably see the same thing. Jaeger logs just each 1000th request so it might look that it does not work. I suspect that configuration for Jaeger defined in microprofile-config.properties is not applied. 

My microprofile-config.properties looks like (taken from [1]):
```
jaeger.service-name=test-traced-service
jaeger.sampler.param=1
jaeger.sampler.type=const
```
[1] https://docs.thorntail.io/4.0.0-SNAPSHOT/#component-opentracing-jaeger

**Comment by zhantemirov** [comment]
Still having issues with discovering traces in Jagger UI. I've used your suggested configuration and it seems to be working, few interesting things are happening though. 

I decided to investigate the issue by starting a WildFly instance, deploying the same WAR archive (that I've been using in migrated tests) and starting a Jaeger (all steps manually). Here are the results:

starting Jaeger -> starting WildFly + WAR deploy -> trace records are presented
starting WildFly + WAR deploy -> starting Jaeger -> records are NOT presented
starting Wildfly  -> starting Jaeger -> WAR deploy -> trace records are presented

I don't know why the second option doesn't work. In my migrated tests the third option seems to not be working, giving it a little bit of time - but I need to do some work on RFE, so maybe filing an issue here would be a temporary option.

**Comment by rsvoboda** [comment]
+1 for issue, can you try it on WF 18 too ? OpenTracing is already integrated so we should file issue against WFLY project.

I assume you hit the app endpoint at the time of last `->`.

Is there a way to figure out Jaeger availability from management model ?

**Comment by zhantemirov** [comment]
@rsvoboda after some additional analysis and documentation reading I managed to identify the  possible source of the issue. 

As mnovak has mentioned before, the reason why the entries are not presented at Jaeger database is that injected Jaeger tracer instance doesn't respect the _META-INF/microprofile-config.properties_ configuration.

As soon as I replace the injected tracer instance by my own that is manually configured by parameters from _microprofile-config.properties_ - traced entries are shown up in Jaeger UI.

So the question is: do we need to tweak migrated tests by this way? Own configured tracer instance logs the entries in a little another way - so the test checks have to be modified too. 

FYI the thorntail OpenTracing tests has a support of Swarm Jaeger tracer helper classes that set up all the necessary Jaeger configuration parameters as well. The problem is (as it is written above) 

> injected Jaeger tracer instance doesn't respect the _META-INF/microprofile-config.properties_ configuration.

WDYT?

**Comment by rsvoboda** [comment]
Let me rephrase the question:

Customer has deployment with META-INF/microprofile-config.properties entries for Jaeger configuration
 - Is Jaeger client configured properly in Thorntail ?
 - Is Jaeger client configured properly in WildFly ?

**Comment by zhantemirov** [comment]
An issue has been filed - https://issues.redhat.com/browse/WFLY-12842

**Comment by mnovak1** [comment]
@zhantemirov hi, can it be marked as done and closed?

**Comment by zhantemirov** [comment]
As it was written in the previous commentary, a JIRA [1] has been filed due to the issue [2] we hit. 

The original intent of this migration were additional OpenTracing tests (http path with wild cards, negative scenarios). These tests are already merged [3] in WildFly master.

Closing the issue.

[1] - https://issues.redhat.com/browse/WFLY-12842
[2] - https://github.com/jboss-eap-qe/eap-microprofile-test-suite/issues/10#issuecomment-559480055
[3] - https://github.com/wildfly/wildfly/pull/12836

**Comment by rsvoboda** [comment]
@mnovak1 shouldn't this TS have opentracing module too ?

At least just with README saying tests are in WF repo + link to the module with tests

tbh I didn't expect everything will end up in WF TS and nothing here.

**Comment by zhantemirov** [comment]
Reopening - will prepare PR for the case when https://issues.redhat.com/browse/WFLY-12842 will be fixed.
